ui/src/app/services/sign-in.service.ts

```diff
@@ -0,0 +1,113 @@
+import 'rxjs/Rx';
+import {Injectable, NgZone} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {User} from 'app/models/user';
+
+declare const gapi: any;
+
+interface BasicProfile {
+  id: string;
+  name: string;
+  givenName: string;
+  familyName: string;
+  imageUrl: string;
+  email: string;
+}
+
+export interface SignInDetails {
```

Comment:
We should decide what we want to treat as the source of truth, to be sure. Right now I think it's useful to reflect what we're getting back from Google...